c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -1,22 +1,605 @@
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(aes256),
+                 byEncryption({Encryption::AES256, Encryption::AES256GCM}));
+    aliases.insert({AES256, aes256});
+    // cipher suites using either 128 or 256 bit AES.
+    auto aes(aes128);
+    aes.insert(aes.end(), aes256.begin(), aes256.end());
+    aliases.insert({AES, aes});
+
+    // AES in Galois Counter Mode (GCM).
+    std::vector<Cipher> aesgcm;
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(aesgcm),
+                 byEncryption({Encryption::AES128GCM, Encryption::AES256GCM}));
+    aliases.insert({AESGCM, aesgcm});
+
+    // Cipher suites using ChaCha20.
+    std::vector<Cipher> chacha20;
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(chacha20),
+                 byEncryption(Encryption::CHACHA20POLY1305));
+    aliases.insert({CHACHA20, chacha20});
+
+    // Cipher suites using triple DES.
+    std::vector<Cipher> triple_des;
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(triple_des),
+                 byEncryption(Encryption::TRIPLE_DES));
+    aliases.insert({TRIPLE_DES, triple_des});
+
+    // Cipher suites using SHA1.
+    std::vector<Cipher> sha1;
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(sha1),
+                 byMessageDigest(MessageDigest::SHA1));
+    aliases.insert({SHA1, sha1});
+    aliases.insert({SHA, sha1});
+    // Cipher suites using SHA256.
+    std::vector<Cipher> sha256;
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(sha256),
+                 byMessageDigest(MessageDigest::SHA256));
+    aliases.insert({SHA256, sha256});
+    // Cipher suites using SHA384.
+    std::vector<Cipher> sha384;
+    std::copy_if(this->all.begin(), this->all.end(), std::back_inserter(sha384),
+                 byMessageDigest(MessageDigest::SHA384));
+    aliases.insert({SHA384, sha384});
+
+    // COMPLEMENTOFDEFAULT:
+    // The ciphers included in ALL, but not enabled by default. Currently this includes all RC4 and
+    // anonymous ciphers. Note that this rule does not cover eNULL, which is not included by ALL
+    // (use COMPLEMENTOFALL if necessary). Note that RC4 based cipher suites are not supported by
+    // Fastly and the only supported anonymous ciphers are `ecdh` and `triple_des`.
+    auto complementOfDefault(ecdh);
+    complementOfDefault.insert(complementOfDefault.end(), triple_des.begin(), triple_des.end());
+    aliases.insert({COMPLEMENTOFDEFAULT, complementOfDefault});
+
+    // The content of the default list is determined at compile time and normally corresponds to
+    // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+    aliases.insert({DEFAULT, parse("ALL:!COMPLEMENTOFDEFAULT:!eNULL")});
+  }
+
+  std::vector<Cipher> parse(std::string_view expression) {
+    /**
+     * All ciphers by their openssl alias name.
+     */
+    auto elements = splitCipherSuiteString(std::string(expression));
```

Comment:
This is allocating a temporary `std::string` as a copy of `expression` that will be freed after the call finishes. The resulting string views in `elements` will be references to the freed memory.
```suggestion
    auto elements = splitCipherSuiteString(expression);
```